src/com/dotcms/rest/api/v1/content/ContentTypeHelper.java

```diff
@@ -126,4 +140,56 @@ private Long getLanguageId (final String userLocaleString, final LanguageAPI lan
         return languageId;
     }
 
+    /**
+     * Return  a {@link List} of BaseContentTypesView
+     *
+     * @param request
+     * @return
+     * @throws DotDataException
+     * @throws LanguageException
+     */
+    public List<BaseContentTypesView> getTypes(HttpServletRequest request ) throws DotDataException, LanguageException {
+        final InitDataObject initData = this.webResource.init(null, true, request, true, null); // should logged in
+        final User user = initData.getUser();
+
+        List<Structure> structures = this.structureAPI.find(user, false, true);
+        List<BaseContentTypesView> result = list();
+
+        if (null != structures) {
+            Locale locale = LocaleUtil.getLocale(request);
+            Map<String, String> baseContentTypeNames = this.getBaseContentTypeNames(locale);
+            BaseContentTypesViewCollection baseContentTypesViewCollection = new BaseContentTypesViewCollection();
+
+            structures.stream()
+                    .forEach(structure -> {
+                        baseContentTypesViewCollection.add(structure, new ContentTypeView(
+                                Structure.Type.getType(structure.getStructureType()).name(),
+                                structure.getName(), structure.getInode(),
+                                this.getActionUrl(request, structure, user)));
+                    });
+
+            result = baseContentTypesViewCollection.getStructureTypeView(baseContentTypeNames);
+        }
+
+        return result;
+    }
+
+    /**
+     * Get Str Type Names
+     * @param locale {@link Locale}
+     * @return Map (type Id -> i18n value)
+     * @throws LanguageException
+     */
+    public final Map<String, String> getBaseContentTypeNames(final Locale locale) throws LanguageException {
+
+        return map(
+                Structure.Type.getType(Structure.Type.CONTENT.getType()).name(), LanguageUtil.get(locale, "Content"),
```

Comment:
If this is call a lot of time, should be a constant.
I mean the Content, Widget, etc, strings